flake.nix

```diff
@@ -33,9 +35,31 @@
       # Use the expression from Nixpkgs instead of rewriting it here.
       sile = pkgs.sile.overrideAttrs(oldAttr: rec {
         version = "${(pkgs.lib.importJSON ./package.json).version}-${version_rev}-flake";
-        src = builtins.filterSource
-          (path: type: type != "directory" || baseNameOf path != ".git")
-        ./.;
+        src = nix-filter.lib {
+          root = ./.;
+          exclude = [
+            # Nix files
+            "flake.nix"
+            "flake.lock"
+            "default.nix"
+            "shell.nix"
+            # git commit and editing format files
+            ".commitlintrc.yml"
+            "package.json"
+            ".husky"
+            ".editorconfig"
+            # CI files
+            ".cirrus.yml"
+            "action.yml"
+            "azure-pipelines.yml"
+            "Dockerfile"
+            # Git files
+            ".github"
+            ".gitattributes"
+            ".gitignore"
+            ".git"
```

Comment:
`.git` is not itself tracked by Git. That makes me wonder, do we need to list other things here that might exist as artifacts in a working copy for one reason or another, or are we starting with just a list of *tracked* files here?

Reply:
The exclude list I wrote here is a list of files not needed for compilation. I think it will be appropriate to also exclude generated files or dist_files to make the source even cleaner in case a real developer (not me) is using `nix build`.